block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagKeyValueStorage.scala

```diff
@@ -248,11 +249,11 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
   override def accessEquivocationsTracker[A](f: EquivocationsTracker[F] => F[A]): F[A] =
     lock.withPermit(f(KeyValueStoreEquivocationsTracker))
 
-  def addFinalizedBlockHash(blockHash: BlockHash): F[Unit] =
-    blockMetadataIndex.addFinalizedBlock(blockHash)
+  def addFinalizedBlockHashes(hashes: List[BlockHash]): F[Unit] =
```

Comment:
I made this accept list because indirectly finalized blocks are processed always in batches. And its better to have it atomic update on KVStore level.